### PR TITLE
test: valid is always bool [run_process_replay]

### PR DESCRIPTION
**Pull request by Qazalin**
Prevents the Linearizer from rendering NumNode(0) and NumNode(1) for valids. Now all non-bool gates are asserted in the UOps type checker.

### CODE REVIEW

**Comment by geohot** [comment]
Want to add an assert to the UOps type checker that confirms they are always bool?

**Comment by Qazalin** [comment]
yea this added it https://github.com/tinygrad/tinygrad/pull/5061

if the uops rewriter is working correctly it should never hit this but looks like it does in multithreading retinanet.

**Comment by geohot** [comment]
Yea this should definitely be fixed in the Linearizer, but also...why isn't the rewriter working?

**Comment by reddyn12** [comment]
> but looks like it does in multithreading retinanet.

I didn't root cause, but multitreading on single gpu worked. Multithreading on multi gpu hit that assert. Try running train with TEST=1 on multigpu to see if multithreading is a problem

**Comment by Qazalin** [comment]
@reddyn12 the dataset is still downloading for me, wanna try running it on this branch?

**Comment by reddyn12** [comment]
I don't have access to a multigpu setup rn, but TEST=1 doesn't need any dataset, uses fake data

File: tinygrad/codegen/linearizer.py [hunk 1]
```diff
@@ -141,7 +141,8 @@ def global_load(self, i:int, idxs:List[Node], acc:Optional[LazyOp]=None, barrier
     invalid_value = 0
     acc_count = 0
     for idx, valid, rep_idx in zip(e_idxs, e_valids, iter_idxs(expand_vars)):
-      this_const, idx, valid = (invalid_value, NumNode(0), NumNode(1)) if valid.max == 0 else (const, idx, valid)
+      this_const, idx = (invalid_value, NumNode(0)) if valid.max == 0 else (const, idx)
+      valid_uop = UOp.const(dtypes.bool, valid.b) if valid.min == valid.max else valid.render(render_ops, self.loop_uops)
```
Comment (Qazalin) [inline tinygrad/codegen/linearizer.py:145]: This felt like a better check, if it renders a non-bool ALU as valid it should be fixed in symbolic.

**Pull request merged**